src/framework/compareConstraints.ml

```diff
@@ -158,7 +158,7 @@ struct
   include PrecCompare.MakeHashtbl (Var) (Dom) (VH)
 end
 
-module CompareEqSys (Sys: EqConstrSys) (VH: Hashtbl.S with type key = Sys.Var.t) =
+module CompareEqSys (Sys: DemandEqConstrSys) (VH: Hashtbl.S with type key = Sys.Var.t) =
```

Comment:
I'm only now thinking about this, but it seems like perhaps here we could avoid the demand: the comparison of solutions shouldn't need to invoke the right-hand sides at all.

Reply:
Oh I misunderstood your point, I thought you wanted to factor it out completely. 

This is of course simple 600c417c